tests/modules/test_morpheus_modules_with_pipeline.py

```diff
@@ -0,0 +1,198 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+
+import cudf
+
+# pylint: disable=unused-import
+import morpheus.modules  # noqa: F401
+from morpheus import messages
+from morpheus.utils.module_ids import FILTER_CM_FAILED
+
+from .morpheus_module_test_pipeline import MorpheusModuleTestPipeline
+
+PACKET_COUNT = 5
+PACKETS_RECEIVED = 0
+
+
+def test_payload_loader_module():
+    df = cudf.DataFrame({
+        'col1': [1, 2, 3, 4, 5],
+        'col2': [1.1, 2.2, 3.3, 4.4, 5.5],
+        'col3': ['a', 'b', 'c', 'd', 'e'],
+        'col4': [True, False, True, False, True]
+    })
+
+    def gen_data():
+        payload = messages.MessageMeta(df)
+        for _ in range(PACKET_COUNT):
+            config = {"tasks": [{"type": "load", "properties": {"loader_id": "payload", "strategy": "aggregate"}}]}
+            msg = messages.ControlMessage(config)
+            msg.payload(payload)
+            yield msg
+
+    def on_next(control_msg):
+        # pylint: disable=global-statement
+        global PACKETS_RECEIVED
+        PACKETS_RECEIVED += 1
```

Comment:
Dont use global values for test assertions like this. You can define `PACKETS_RECEIVED` inside of the function and then use `nonlocal PACKETS_RECEIVED` inplace of `global PACKETS_RECEIVED`. Then each function will have a variable scoped correctly

Reply:
This should be applied to all tests